pkg/globalmanager/controllers/jointinference/jointinferenceservice.go

```diff
@@ -422,113 +351,234 @@ func (c *Controller) createWorkers(service *sednav1.JointInferenceService) (acti
 	if err != nil {
-		return fmt.Errorf("failed to get edge model %s: %w",
-			edgeModelName, err)
+		if errors.IsNotFound(err) {
+			// If the service does not exist, create a new one and retrieve its address
+			klog.Info("Cloud service not found, creating new service...")
+			bigModelHost, err = runtime.CreateEdgeMeshService(c.kubeClient, oldService, jointInferenceForCloud, BigModelPort)
+			if err != nil {
+				klog.Errorf("Failed to create EdgeMesh service for service %s/%s: %v", oldService.Namespace, oldService.Name, err)
+			}
+		} else {
+			klog.Errorf("Failed to get cloud service %s/%s: %v", oldService.Namespace, oldService.Name, err)
+		}
+	} else {
+		bigModelHost = fmt.Sprintf("%s.%s", svc.Name, svc.Namespace)
 	}
 
-	secretName := edgeModel.Spec.CredentialName
-	var modelSecret *v1.Secret
-	if secretName != "" {
-		modelSecret, _ = c.kubeClient.CoreV1().Secrets(service.Namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	// Check if the edge worker configuration has changed
+	if !reflect.DeepEqual(oldService.Spec.EdgeWorker, newService.Spec.EdgeWorker) {
+		// Update edge worker and log any errors
+		if err := c.updateEdgeWorker(newService, bigModelHost); err != nil {
+			klog.Errorf("Failed to update edge worker for service %s/%s: %v", newService.Namespace, newService.Name, err)
+		}
 	}
 
-	edgeWorker := service.Spec.EdgeWorker
-	HEMParameterJSON, _ := json.Marshal(edgeWorker.HardExampleMining.Parameters)
-	HEMParameterString := string(HEMParameterJSON)
+	return nil
+}
 
+func (c *Controller) createOrUpdateWorker(service *sednav1.JointInferenceService, workerType string, bigModelHost string, bigModelPort int32, create bool) error {
+	var modelName string
+	var modelTemplate v1.PodTemplateSpec
 	var workerParam runtime.WorkerParam
 
+	// Set the corresponding parameters according to the workerType.
+	switch workerType {
+	case jointInferenceForCloud:
+		modelName = service.Spec.CloudWorker.Model.Name
+		modelTemplate = *service.Spec.CloudWorker.Template.DeepCopy()
+
+		workerParam.Env = map[string]string{
+			"BIG_MODEL_BIND_PORT": strconv.Itoa(int(bigModelPort)),
+		}
+		workerParam.WorkerType = workerType
+		workerParam.HostNetwork = false // The cloud does not need HostNetwork.
+
+	case jointInferenceForEdge:
+		modelName = service.Spec.EdgeWorker.Model.Name
+		modelTemplate = *service.Spec.EdgeWorker.Template.DeepCopy()
+
+		HEMParameterJSON, _ := json.Marshal(service.Spec.EdgeWorker.HardExampleMining.Parameters)
+		HEMParameterString := string(HEMParameterJSON)
+
+		workerParam.Env = map[string]string{
+			"BIG_MODEL_IP":   bigModelHost,
+			"BIG_MODEL_PORT": strconv.Itoa(int(bigModelPort)),
+			"HEM_NAME":       service.Spec.EdgeWorker.HardExampleMining.Name,
+			"HEM_PARAMETERS": HEMParameterString,
+
+			"LC_SERVER": c.cfg.LC.Server,
+		}
+		workerParam.WorkerType = workerType
+		workerParam.HostNetwork = true // Edge nodes need HostNetwork.
```

Comment:
ditto

Reply:
After these pull requests are merged, the current joint-inference examples need to add the hostNetwork configuration in the YAML, similar to the following:
```yaml
CLOUD_NODE="cloud-node-name"
EDGE_NODE="edge-node-name"


kubectl create -f - <<EOF
apiVersion: sedna.io/v1alpha1
kind: JointInferenceService
metadata:
  name: helmet-detection-inference-example
  namespace: default
spec:
  edgeWorker:
    model:
      name: "helmet-detection-inference-little-model"
    hardExampleMining:
      name: "IBT"
      parameters:
        - key: "threshold_img"
          value: "0.9"
        - key: "threshold_box"
          value: "0.9"
    template:
      spec:
        nodeName: $EDGE_NODE
        dnsPolicy: ClusterFirstWithHostNet
        hostNetwork: true
        containers:
        - image: kubeedge/sedna-example-joint-inference-helmet-detection-little:v0.3.0
          imagePullPolicy: IfNotPresent
          name:  little-model
          env:  # user defined environments
          - name: input_shape
            value: "416,736"
          - name: "video_url"
            value: "rtsp://localhost/video"
          - name: "all_examples_inference_output"
            value: "/data/output"
          - name: "hard_example_cloud_inference_output"
            value: "/data/hard_example_cloud_inference_output"
          - name: "hard_example_edge_inference_output"
            value: "/data/hard_example_edge_inference_output"
          resources:  # user defined resources
            requests:
              memory: 64M
              cpu: 100m
            limits:
              memory: 2Gi
          volumeMounts:
            - name: outputdir
              mountPath: /data/
        volumes:   # user defined volumes
          - name: outputdir
            hostPath:
              # user must create the directory in host
              path: /joint_inference/output
              type: Directory

  cloudWorker:
    model:
      name: "helmet-detection-inference-big-model"
    template:
      spec:
        nodeName: $CLOUD_NODE
        dnsPolicy: ClusterFirstWithHostNet
        containers:
          - image: kubeedge/sedna-example-joint-inference-helmet-detection-big:v0.3.0
            name:  big-model
            imagePullPolicy: IfNotPresent
            env:  # user defined environments
              - name: "input_shape"
                value: "544,544"
            resources:  # user defined resources
              requests:
                memory: 2Gi
EOF
```